libtiledbvcf/src/utils/utils.cc

```diff
@@ -557,6 +558,22 @@ const std::string& version_info() {
   return version;
 }
 
+#ifdef _WIN32
+#include <windows.h>
+#endif
+
+uint32_t system_memory_mb() {
```

Comment:
This is only used for the default setting so it is acceptable. Please note however that this detection does not work for cgroup restricted envs. I.e if you run inside docker and limit the docker container to certain memory we'll default to too large a value. I don't think this is normal behavior since users are likely to be setting the memory limit themselves. I just want to mention it for general awareness.

Reply:
Good point. Knowing the cgroup limit would be useful, then we could return the min of the cgroup limit and available system memory.

For one approach, the cgroup/cgroup2 memory limits can be discovered through these files (on Linux):
```
/proc/mounts
/proc/<pid>/cgroup
# cgroup
<path-to-cgroup-memory-controller-mount>/<pid-cgroup-path>/memory.limit_in_bytes
# cgroup2
<path-to-cgroup2-mount>/<pid-cgroup-path>/memory.max
```